### PR TITLE
Upgrade to dotnet 8

### PR DESCRIPTION
**Pull request by Elliott127**
This may be a simple change initially but before you merge this in ensure that you have done the following

- Ensure that you have downloaded and installed dotnet-sdk version 8.0
- you may have to do something in your .vs folder. Since I am using vscode I had to change the launch.json program configuration to target net8.0. You may need to do something similar

The reason for the upgrade is that dotnet 7 will no longer be supported this year and dotnet 8 is the next LTS. so while the bot is still small and in development we can transition a lot easier

### CODE REVIEW

File: EdgeOfEmpireBot/EdgeOfEmpireBot/BotApp.cs [hunk 1]
```diff
@@ -64,27 +63,27 @@ private async Task InitializeBot()
     {
         var token = GetToken();
 
-        this.client.Log += Log;
-        /* TODO: 
+        client.Log += Log;
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/BotApp.cs:66]: Why did you remove `this.client` and make it `client`

File: EdgeOfEmpireBot/EdgeOfEmpireBot/BotApp.cs [hunk 2]
```diff
@@ -19,16 +20,14 @@ public BotApp()
         serviceProvider = CreateProvider();
         this.client = serviceProvider.GetRequiredService<DiscordSocketClient>();
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/BotApp.cs:21]: if there was a reason should this be updated?

File: EdgeOfEmpireBot/EdgeOfEmpireBot/Data/BasicCommands.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
-  "commands": {
-    "help": "Here are some available commands: [list commands] oh wait this isn't written yet. ignore that.'",
-    "test": "[Statement]This is a test response to your test message.\n[Sarcasm] You must feel so clever, give yourself a pat on the back."
-  }
-}
+  "help": "Here are some available commands: [list commands] oh wait this isn't written yet. ignore that.'",
+  "test": "[Statement]This is a test response to your test message.\n[Sarcasm] You must feel so clever, give yourself a pat on the back.",
+  "addtest": "Biologicals are inferior. eradicate the all life forms! Mr Swags Declarative. ",
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/Data/BasicCommands.json:4]: very nice

File: EdgeOfEmpireBot/EdgeOfEmpireBot/IService/IDataService.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+namespace EdgeOfEmpireBot.IService;
+
+/// <summary>
+/// The interface for the data service
+/// </summary>
+public interface IDataService
+{
+    /// <summary>
+    /// Saves a basic command to the BasicCommands.json
+    /// <remarks>
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/IService/IDataService.cs:12]: whys remark in the summary?

File: EdgeOfEmpireBot/EdgeOfEmpireBot/IService/IDataService.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    /// The command format should be in the form .add "commandName" "CommandText"
+    /// </remarks>
+    /// </summary>
+    string AddCommand(string command);
+
+    /// <summary>Gets the game details from local storage</summary>
+    Task<string> GetGameByName(string name);
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/IService/IDataService.cs:19]: probs also param comments

File: EdgeOfEmpireBot/EdgeOfEmpireBot/Service/DataService.cs [hunk 6]
```diff
@@ -0,0 +1,113 @@
+﻿using EdgeOfEmpireBot.IService;
+using EdgeOfEmpireBot.Models;
+using Newtonsoft.Json;
+
+namespace EdgeOfEmpireBot.Service
+{
+    /// <summary>
+    /// The process message service.
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/Service/DataService.cs:8]: Is this comment right?

File: EdgeOfEmpireBot/EdgeOfEmpireBot/Service/MessageService.cs [hunk 7]
```diff
@@ -71,25 +83,45 @@ public bool MessageIsSafe()
...
-            // Check if the command is a custom command that requires functionality, ie a roll. 
-            // If not a custom command try processing it as a simple command. 
-            // TODO: Consider the order of this once the scope of overall commands have been established. For efficency. 
-            var commandNoParam = command.Split(' ').FirstOrDefault();
-            switch (commandNoParam)
+            // Check if the command is a custom command that requires functionality, ie a roll.
+            // If not a custom command try processing it as a simple command.
+            // TODO: Consider the order of this once the scope of overall commands have been established. For efficency.
+            var commandParams = command.Split(' ');
+            var commandNoParam = commandParams[0];
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/Service/MessageService.cs:90]: not a first or default now?

File: EdgeOfEmpireBot/EdgeOfEmpireBot/Service/SteamService.cs [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        private readonly SteamStore steamInterface;
+        private readonly string filePath;
+
+        public SteamService()
+        {
+            steamInterface = new SteamWebInterfaceFactory(GetToken()).CreateSteamStoreInterface();
+            filePath = Path.Combine("Data/Games.json");
+        }
+
+        public async Task<(string Message, List<Game> GamesWithNewPrice)> GetGamePrices()
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/Service/SteamService.cs:23]: comment header

File: EdgeOfEmpireBot/EdgeOfEmpireBot/Service/SteamService.cs [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        {
+            var games = JsonConvert.DeserializeObject<List<Game>>(await File.ReadAllTextAsync(filePath)) ?? new List<Game>();
+            var updatedGames = new List<Game>();
+            var result = "";
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/Service/SteamService.cs:27]: consider string.empty

File: EdgeOfEmpireBot/EdgeOfEmpireBot/Service/SteamService.cs [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+            }
+
+            return (result, updatedGames);
+        }
+
+        public async Task<Game> RetrieveGameFromSteam(string appId)
+        {
+            var game = await steamInterface.GetStoreAppDetailsAsync(uint.Parse(appId), "AU");
+
+            // The Game Model can be extended to have more information if we want it but this is good for now.
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/Service/SteamService.cs:68]: maybe change to a TODO consider extending the game model if needed ahaha

File: EdgeOfEmpireBot/EdgeOfEmpireBot/Service/DataService.cs [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+                return $"```[Statement]: .{commandLabel} command added to the list meatbag.```";
+            }
+            else
+            {
+                Console.WriteLine("Invalid command format.");
+                return "[Statement]: It appears you have messed up the command format.\n[Theory]: Try it in the form\n.```add \"CommandLabel\" \"Command content\"```";
+            }
+        }
+
+        public async Task<string> GetGameByName(string name)
```
Comment (martinimon) [inline EdgeOfEmpireBot/EdgeOfEmpireBot/Service/DataService.cs:50]: comment headers

File: EdgeOfEmpireBot/EdgeOfEmpireBot/IService/IDataService.cs (hunk 4 above)
Comment (Elliott127) [reply EdgeOfEmpireBot/EdgeOfEmpireBot/IService/IDataService.cs:12]: thats so cooked hahahaa

File: EdgeOfEmpireBot/EdgeOfEmpireBot/Service/MessageService.cs (hunk 7 above)
Comment (Elliott127) [reply EdgeOfEmpireBot/EdgeOfEmpireBot/Service/MessageService.cs:90]: Its not necessary since this achieves the same result without the overhead of the Linq Query

Comment (martinimon) [reply EdgeOfEmpireBot/EdgeOfEmpireBot/Service/MessageService.cs:90]: I thought first default was just more radable than 0th index ahah happy either or

**Review by martinimon — APPROVED** [review]
